hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/FlinkOptions.java

```diff
@@ -725,40 +725,45 @@ private FlinkOptions() {
       .withDescription("Target IO in MB for per compaction (both read and write), default 500 GB");
 
   public static final ConfigOption<Boolean> CLEAN_ASYNC_ENABLED = ConfigOptions
-      .key("clean.async.enabled")
+      .key("hoodie.cleaner.async.enabled")
```

Comment:
`clean.async.enabled` is shortcut option specifically for flink, `hoodie.cleaner.async.enabled` should be the fallback key instead.

Reply:
@danny0405 , if it doesn't bother you, could you, please, clarify, why Flink options are different from the similar Hudi kernel options? From user point of view, it's unnecessary complication, you need to look for Flink variation of naming even if you know some Spark or kernel naming version for the same option.

Could we create some naming convention for options? It's a nightmare now, as I mentioned in [HUDI-5738](https://issues.apache.org/jira/browse/HUDI-5738), we have 900+ options for now.

Reply:
There are some parameters without shortcuts in `FlinkOptions`:
```
hoodie.bucket.index.hash.field
hoodie.bucket.index.num.buckets
hoodie.database.name
hoodie.datasource.merge.type
hoodie.datasource.query.type
hoodie.datasource.write.hive_style_partitioning
hoodie.datasource.write.keygenerator.class
hoodie.datasource.write.keygenerator.type
hoodie.datasource.write.partitionpath.field
hoodie.datasource.write.partitionpath.urlencode
hoodie.datasource.write.recordkey.field
hoodie.index.bucket.engine
hoodie.table.name
```

Reply:
I kind of think the sql options should not be too long, and we keep compatibility for hudi altenatives.

Reply:
Ok, I see your point. Modified corresponding commit by saving previous keys in `FlinkOptions` and adding new keys as fallback keys. Rebased on the current master and force pushed.